File: Seminar5/HomeWork/Controller/PresenterRation.java
package HomeWork.Controller;

import HomeWork.Model.Rational.ModelRational;
import HomeWork.View.ViewRational;

public class PresenterRation {
    ViewRational view;
    ModelRational model;

    public PresenterRation(ModelRational m, ViewRational v) {
        model = m;
        view = v;
    }

    public PresenterRation() {

    }

    public void buttonClick() {
        double a = view.getValue("a: ");
        double b = view.getValue("b: ");
        model.setX(a);
        model.setY(b);

        double result = model.result();
        view.print(result, "Result: ");
    }

}

File: FinalTaskCalculator/Controller/PresenterRation.cs

using static ModelRational.Rational;
using static View.ViewRational;

namespace Controller
{


    public class PresenterRation
    {
        View.ViewRational view = null!;
        ModelRational.Rational model = null!;

        public PresenterRation(ModelRational.Rational m, View.ViewRational v)
        {
            model = m;
            view = v;
        }

        public PresenterRation()
        {

        }

        public void ButtonClick()
        {
            double a = view.GetValue("a: ");
            double b = view.GetValue("b: ");
            model.SetX(a);
            model.SetY(b);

            double result = model.Result();
            view.Print(result, "Result: ");
        }

    }
}